course_project/23/src/main/java/com/example/course_project/ConnectionRepository.java

```diff
@@ -0,0 +1,20 @@
+package com.example.course_project;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+
+public abstract class ConnectionRepository {
+    protected Connection connection;
+
+    protected ConnectionRepository() {
+        var url = System.getenv("DATABASE_URL");
```

Comment:
This is not a bad idea at all, but it's a bit better to use application.properties/application.yml for the env vars